crates/proof-of-sql/src/sql/proof/query_proof.rs

```diff
@@ -56,7 +56,7 @@ impl<CP: CommitmentEvaluationProof> QueryProof<CP> {
         let mut result_builder = ResultBuilder::new(table_length);
         let result_cols = expr.result_evaluate(&mut result_builder, &alloc, accessor);
         let provable_result =
-            ProvableQueryResult::new(&result_builder.result_index_vector, &result_cols);
+            ProvableQueryResult::new(result_builder.table_length() as u64, &result_cols);
```

Comment:
I feel like this is wrong. The result doesn't have the same length as the original data. Why isn't it panicking?

Reply:
@JayWhite2357 It is not panicking because currently `result_evaluate` (line 57) will set `table_length` to be the output table length.

After the next PR we will differentiate between input and output table lengths better.

Reply:
Maybe we should make it very clear in `provable_query_results.rs` etc that we always use `output_length`, that is, actually rename all the variables. Meanwhile if we use input length(s) we should label them as such.

Reply:
BTW `result_builder.result_index_vector` serves the exact same purpose since all that it can be in `ProofPlan`s that do exist is `Indexes::Dense((0..output_length))`